packages/astro/src/vite-plugin-scanner/scan.ts

```diff
@@ -82,5 +85,16 @@ export async function scan(
 		}
 	}
 
+	if (settings && fileURL) {
+		const route: RouteOptions = {
+			component: rootRelativePath(settings.config.root, fileURL, false),
+			prerender: pageOptions.prerender,
+		};
+		for (const handler of settings.routeOptionsHandlers) {
+			handler(route);
+		}
```

Comment:
I have a concern about this approach, where two or more integrations can change `prerender` as they want for the same route, and only the last one wins. I would use a different approach: make the handler **return** the `prerender` value, and log that "X integration changed to `prerender` value to Y".  

Plus, I would add a warning in case an integration overrides a value that was previously set by another integration.

Reply:
Would love to hear what others think about this too 🤔 The reason I made "last one wins" without warnings is because that's also how the rest of the Astro config / Vite config works. Also would there be a way to suppress the warning if the user thinks it's fine?

Reply:
Mutating global configuration is what the integration hooks are mostly about, so I don't think this is in different in that regard. Multiple integrations could make changes to config (like `outDir` comes to mind) that conflict with each other. We don't have any real mechanisms for preventing this, and I'm not sure how we really could. So to me this is just another extension of that same concept.

Adding some dev warnings about conflicts wouldn't be a bad thing though if that helps alleviate the concern.

Reply:
I added a debug log if the value is changed more than once